encryption/encryption.cc

```diff
@@ -265,6 +266,16 @@ Status KeyManagedEncryptedEnv::NewSequentialFile(
     case EncryptionMethod::kAES192_CTR:
     case EncryptionMethod::kAES256_CTR:
       s = encrypted_env_->NewSequentialFile(fname, result, options);
+      if (!s.ok()) {
+        return s;
+      }
+      // Hack: sometimes the current file may be read as an encrypted
+      // file by mistakes.
+      if (ShouldSkipEncryption(fname) && !isValidCurrentFile(result)) {
+        s = target()->NewSequentialFile(fname, result, options);
+      } else {
+        s = encrypted_env_->NewSequentialFile(fname, result, options);
+      }
```

Comment:
Let's change the flow here:
```
if (s.ok() && ShouldSkipEncryption(fname)) {
  if (IsValidCurrentFile(std::move(result)) {
    s = target()->NewSequentialFile(...);
  } else {
    s = encrypted_env_->NewSequentialFile(...);
  }
}
```
basically I would prefer avoid early return, since other cases within the switch...case... block does not immediately return.